src/CoreBundle/Bridge/Chameleon/Module/Sidebar/SidebarBackendModule.php

```diff
@@ -0,0 +1,257 @@
+                $cmsModule->fieldIconFontCssClass,
+                $this->getModuleTargetUrl($cmsModule)
+            );
+        }
+    }
+
+    private function isModuleAccessAllowed(\TCMSUser $activeUser, \TdbCmsModule $cmsModule): bool
+    {
+        return true === $activeUser->oAccessManager->user->IsInGroups($cmsModule->fieldCmsUsergroupId);
+    }
+
+    private function getModuleTargetUrl(\TdbCmsModule $cmsModule): string
+    {
+        $url = PATH_CMS_CONTROLLER.'?pagedef='.$cmsModule->fieldModule;
+        if ('' !== $cmsModule->fieldParameter) {
+            $url .= '&'.$cmsModule->fieldParameter;
+        }
+        if ('' !== $cmsModule->fieldModuleLocation) {
+            $url .= '&_pagedefType='.$cmsModule->fieldModuleLocation;
+        }
+
+        return $url;
+    }
+
+    private function mergeMenuItems(array $menuItemsRaw): array
+    {
+        $categoryNames = $this->getCategoryNames();
+        $menuItems = [];
+        foreach ($menuItemsRaw as $categoryId => $items) {
+            if (true === \array_key_exists($categoryId, $categoryNames)) {
+                $categoryName = $categoryNames[$categoryId];
+            } else {
+                $categoryName = '-';
+            }
+            \usort($items, function (MenuItem $menuItem1, MenuItem $menuItem2) {
+                return \strcmp($menuItem1->getName(), $menuItem2->getName());
+            });
+            $menuItems[] = new MenuCategory($categoryName, $items);
+        }
+        \usort($menuItems, function (MenuCategory $menuCategory1, MenuCategory $menuCategory2) {
+            return \strcmp($menuCategory1->getName(), $menuCategory2->getName());
+        });
+
+        return $menuItems;
+    }
+
+    private function getCategoryNames(): array
+    {
+        $contentBoxList = \TdbCmsContentBoxList::GetList('SELECT * FROM `cms_content_box`');
+        $names = [];
+        while ($contentBox = $contentBoxList->Next()) {
+            $names[$contentBox->id] = $contentBox->fieldName;
+        }
+
+        return $names;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function GetHtmlHeadIncludes()
+    {
+        $includes = parent::GetHtmlHeadIncludes();
+        $includes[] = sprintf('<link rel="stylesheet" href="%s/coreui/css/perfect-scrollbar.css" type="text/css" />',
+            \TGlobal::GetPathTheme());
+
+        return $includes;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function GetHtmlFooterIncludes()
+    {
+        $includes = parent::GetHtmlFooterIncludes();
+        $includes[] = sprintf('<script src="%s" type="text/javascript"></script>',
+            \TGlobal::GetStaticURLToWebLib('/javascript/modules/sidebar/sidebar.js'));
+
+        return $includes;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function DefineInterface()
+    {
+        parent::DefineInterface();
+        $this->methodCallAllowed[] = 'toggleSidebar';
+    }
+
+    public function toggleSidebar(): void
```

Comment:
This is not so much (active) toggle but "save toggle state in session"?

Reply:
From the server's point of view this is a toggle (whenever the sidebar is loaded after this call, it will have the respective state), so I'd say the name fits.